sys/kern/src/arch/rv64/task.rs

```diff
@@ -34,7 +34,7 @@ pub unsafe fn get_current_task() -> &'static task::Task {
     unsafe {
         asm!("csrr {0}, mscratch", out(reg) task);
         uassert!(task != 0);
-        core::mem::transmute::<usize, &task::Task>(task)
+        &*(task as *const task::Task)
```

Comment:
Does this feel dirty? We're casting task as a const pointer to only deref and take address of it to obtain the pointer value. I know that's what has to be done, I assume obviously.  Not that the original was correct either. Makes me wonder if there's more to teach to the compiler on this one in the long run.

Reply:
> Does this feel dirty? We're casting task as a const pointer to only deref and take address of it to obtain the pointer value. I know that's what has to be done, I assume obviously.

yes, it feels really dirty. I am hoping there is some way to do this in a cleaner way and I am just not aware of it. I have added @picostove and @zachreizner to see if they have any inputs on doing this differently.

> Makes me wonder if there's more to teach to the compiler on this one in the long run.

That is probably what I concluded. But I could be missing something that Rust already provides.

Reply:
The canonical way to turn pointers into references in Rust is to use the weird looking `&*` operators. That's actually two operators (an unsafe Deref followed by a borrow) known as a reborrow: https://doc.rust-lang.org/reference/types/pointer.html#raw-pointers-const-and-mut.